packages/tools/devtools/devtools-browser-extension/src/background/BackgroundScript.ts

```diff
@@ -170,7 +171,10 @@ browser.runtime.onConnect.addListener((devtoolsPort: Port): void => {
 					message,
 				);
 			} else {
-				if (message.source === extensionMessageSource) {
+				if (
+					message.source === extensionMessageSource ||
+					message.source === extensionPopupMessageSource
+				) {
```

Comment:
Nit: Might be worth extracting the expected sources into a variable for documentation purposes. E.g.
```typescript
const allowedMessageSources = [extensionMessageSource, extensionPopupMessageSource];
if (allowedMessageSources.includes(message.source) { ... }
```

Reply:
Same recommendation for other places we do these checks